faker/providers/company/pt_BR/__init__.py

```diff
@@ -62,3 +80,15 @@ def catch_phrase(self):
         catch_phrase = self.generator.parse(pattern)
         catch_phrase = catch_phrase[0].upper() + catch_phrase[1:]
         return catch_phrase
+
+    @classmethod
+    def company_id(cls):
+        digits = random.sample(range(10), 8) + [0, 0, 0, 1]
```

Comment:
instead of the global `random` module, you should use the provider's own random instance, so that it can be frozen. Simply call `cls.random_sample`, instead of `random.sample`. The API is the same